consumer.go

```diff
@@ -181,35 +189,50 @@ func (c *Consumer) ScanShard(
 
 			// loop records of page
 			for _, r := range resp.Records {
-				status := fn(r)
-
-				if !status.SkipCheckpoint {
-					lastSeqNum = *r.SequenceNumber
-
-					if err := c.checkpoint.Set(c.streamName, shardID, lastSeqNum); err != nil {
-						return err
-					}
-				}
-
-				if err := status.Error; err != nil {
+				lastSeqNum, isScanStopped, err = c.handleRecord(shardID, r, fn)
```

Comment:
since we have access to `lastSeqNum` here in the loop; might make the `handleRecord` signature slightly simpler to remove it from the return vals?

```go
for _, r := range resp.Records {
  isScanStopped, err := c.handleRecord(shardID, r, fn)
  if err != nil {
    return err
  }
  if isScanStopped {
    return nil
  }
  lastSeqNum = *r.SequenceNumber // set it here
}
```

Reply:
That's a great idea! I'm going to put your suggestion in the PR. Thank you.